clippy_lints/src/manual_unwrap_or.rs

```diff
@@ -43,32 +43,50 @@ impl LateLintPass<'_> for ManualUnwrapOr {
         if in_external_macro(cx.sess(), expr.span) {
             return;
         }
-        lint_option_unwrap_or_case(cx, expr);
+        lint_manual_unwrap_or(cx, expr);
     }
 }
 
-fn lint_option_unwrap_or_case<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) {
-    fn applicable_none_arm<'a>(arms: &'a [Arm<'a>]) -> Option<&'a Arm<'a>> {
+#[derive(Copy, Clone)]
+enum Case {
+    Option,
+    Result,
+}
+
+impl Case {
+    fn unwrap_fn_path(&self) -> &str {
```

Comment:
I think it would be more idiomatic to implement `Display` for `Case`

Reply:
To display the unwrap fn path? I don't really agree here. A `Case` display string (`Option` for example) and its associated `Option::unwrap` function are different things. I prefer keeping the `unwrap_fn_path` as it conveys more meaning then the more general `fmt()`.

Reply:
I see your point, although I think in that case I would maybe rename the struct :)

But TBH I think it's fine both ways, let's leave it like it is now :+1: